examples/basic_gpio_example.py

```diff
@@ -0,0 +1,111 @@
+from supernovacontroller.sequential import SupernovaDevice
+from BinhoSupernova.commands.definitions import (
+    GpioPinNumber, GpioLogicLevel, GpioFunctionality, GpioTriggerType
+)
+from threading import Event
+gpio_interrupt_event = Event()
+
+def main():
+    """
+    Basic example to illustrate GPIO usage with SupernovaController.
+
+    In this example we initialize GPIO pins with the following configurations:
+        GPIO_6: Digital Output
+        GPIO_5: Digital Input with Interrupts for both rising and falling edges
+        This example assumes pin 5 and pin 6 are short-circuited.
+
+    The script performs the following steps:
+        1. Configures GPIO_6 as a Digital Output.
+        2. Configures GPIO_5 as a Digital Input.
+        3. Sets GPIO_6 to HIGH and reads the value from GPIO_5.
+        4. Sets GPIO_6 to LOW and reads the value from GPIO_5.
+        5. Configures an interrupt on GPIO_5 for both rising and falling edges.
+        6. Toggles GPIO_6 to generate interrupts on GPIO_5.
+        7. Disables the interrupt on GPIO_5.
+    """
+    device = SupernovaDevice()
+
+    print("Opening Supernova host adapter device and getting access to the GPIO interface...")
+    info = device.open()
+    gpio = device.create_interface("gpio")
+
+    print("Initializing GPIO peripheral.")
+    (success, _) = gpio.set_bus_voltage(3300)
+    if not success:
+        print("Couldn't set the GPIO bus voltage.")
+        exit(1)
+
+    print("Configuring GPIO pins...")
+    (success, _) = gpio.configure_pin(GpioPinNumber.GPIO_6, GpioFunctionality.DIGITAL_OUTPUT)
+    if not success:
+        print("Couldn't configure GPIO_6 as Digital Output.")
+        exit(1)
+
+    (success, _) = gpio.configure_pin(GpioPinNumber.GPIO_5, GpioFunctionality.DIGITAL_INPUT)
+    if not success:
+        print("Couldn't configure GPIO_5 as Digital Input.")
+        exit(1)
+
+    print("Setting GPIO_6 to HIGH and reading value from GPIO_5...")
+    (success, _) = gpio.digital_write(GpioPinNumber.GPIO_6, GpioLogicLevel.HIGH)
+    if not success:
+        print("Couldn't set GPIO_6 to HIGH.")
+        exit(1)
+
+    (success, value) = gpio.digital_read(GpioPinNumber.GPIO_5)
+    if success:
+        print(f"GPIO_5 read value: {value}")
+    else:
+        print("Failed to read GPIO_5 value.")
+
+    print("Setting GPIO_6 to LOW and reading value from GPIO_5...")
+    (success, _) = gpio.digital_write(GpioPinNumber.GPIO_6, GpioLogicLevel.LOW)
+    if not success:
+        print("Couldn't set GPIO_6 to LOW.")
+        exit(1)
+
+    (success, value) = gpio.digital_read(GpioPinNumber.GPIO_5)
+    if success:
+        print(f"GPIO_5 read value: {value}")
+    else:
+        print("Failed to read GPIO_5 value.")
+
+    # -- Interruptuions
+    def is_gpio_interrupt(name, message):
+        is_gpio_interruption = message['name'].strip() == "GPIO INTERRUPTION"
+        return message['name'].strip() == "GPIO INTERRUPTION"
```

Comment:
I would return the value of `is_gpio_interruption `

Reply:
I opted for removing the definiton of `is_gpio_interrruption` and just directly returning `message['name'].strip() == "GPIO INTERRUPTION"`.